stuff/cron/testing/test_plagiarism_detector.py

```diff
@@ -0,0 +1,299 @@
+                    (`acl_id`, `title`, `alias`, `commit`, `current_version`)
+                    VALUES
+                    (%s, %s, %s, %s, %s)
+                ''', (
+                    acl_id,
+                    problem_alias,
+                    problem_alias,
+                    problem_commit,
+                    problem_version,
+                ))
+            problem_id = cur.lastrowid
+            cur.execute(
+                '''
+                    INSERT INTO `Problemset_Problems`
+                    (`problemset_id`, `problem_id`, `version`)
+                    VALUES
+                    (%s, %s, %s)
+                ''', (
+                    problemset_id,
+                    problem_id,
+                    problem_version,
+                ))
+            problem_ids.append(problem_id)
+
+        # add submissions to the contest
+        guid = 1
+        no_of_users: int = 3
+        guids: List[str] = []
+        for _ in range(no_of_users):
+            username = ''.join(random.choices(string.ascii_letters, k=20))
+            cur.execute(
+                '''
+                    INSERT INTO `Identities`
+                        (`username`)
+                    VALUES
+                        (%s);
+                ''',
+                (username, ),
+            )
+            identity_id: int = cur.lastrowid
+
+            for problem_id in problem_ids:
+                cur.execute(
+                    '''
+                        INSERT INTO `Submissions`
+                        (`identity_id`,
+                            `problem_id`, `problemset_id`,
+                            `guid`, `language`, `status`,
+                            `verdict`, `type`
+                        )
+                        VALUES
+                        (%s, %s, %s, %s, %s, %s, %s,
+                        %s);
+                    ''', (
+                        identity_id,
+                        problem_id,
+                        problemset_id,
+                        f'{guid:032x}',
+                        "cpp20-gcc",
+                        "ready",
+                        "AC",
+                        "test",
+                    ))
+                guids.append(f'{guid:032x}')
+                guid += 1
+        contest.guids = guids
+    dbconn.conn.commit()
+
+    return contest
+
+
+def test_get_contest(dbconn: lib.db.Connection) -> None:
+    contest = create_contest(dbconn)
+
+    get_contests_detector = cron.plagiarism_detector.get_contests(dbconn)
+
+    contests: List[int] = [c['contest_id'] for c in get_contests_detector]
+
+    assert contest.contest_id in contests
+
+
+def test_submission_ids(dbconn: lib.db.Connection) -> None:
+
+    contest = create_contest(dbconn)
+    submissions = cron.plagiarism_detector.get_submissions_for_contest(
+        dbconn, contest.contest_id)
+    for submission in submissions:
+        assert submission['guid'] in contest.guids
```

Comment:
```suggestion
    guids = set(submission['guid'] for submission in submissions)
    assert guids == set(contest.guids)
```

this is more robust because it also checks that there are no unexpected GUIDs.

Reply:
Got it!!